keyboards/lily58/keymaps/lily58l_vii33/keymap.c

```diff
@@ -0,0 +1,242 @@
+    KC_LCTL, KC_Y,  KC_X,  KC_C,    KC_V,    KC_B,    GMESCN,         KC_F12,  KC_N,     KC_M,   KC_LEFT, KC_UP,  KC_RIGHT, KC_DOWN, 
+                           KC_CAPS, KC_LALT, KC_SPC,  KC_SPC,         KC_ENT,  KC_BSPC,  KC_F7,  KC_DEL
+)
+
+};
+
+/* -------------------------------------------------------------------------------- */
+
+/* Different tapping timings */
+uint16_t get_tapping_term(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case GUI_S:
+        case GUI_L:
+            return TAPPING_TERM + 150;
+        case SHFT_SP:
+            return TAPPING_TERM - 60;
+        case MOD:
+            return TAPPING_TERM - 60;
+        default:
+            return TAPPING_TERM;
+    }
+}
+
+
+/* Tapping Force Hold done per key. Better than general setting. See also https://github.com/qmk/qmk_firmware/blob/master/docs/tap_hold.md */
+bool get_tapping_force_hold(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case GUI_S:
+        case ALT_D:
+        case CTL_F:
+        case CTL_J:
+        case ALT_K:
+        case GUI_L:
+        case SHFT_SP:
+            return true;
+        default:
+            return false;
+    }
+}
+
+
+/* Rotary encoder */
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+  if (index == 0) {             // Encoder on master side
+    if (IS_LAYER_ON(_MOD))          // _MOD LAYER
+    {
+        if ( get_mods() & MOD_MASK_SHIFT ) {
+            if (clockwise) {
+                rgblight_increase_sat_noeeprom();
+            } else {
+                rgblight_decrease_sat_noeeprom();
+            }
+        } else {
+            if (clockwise) {
+                rgblight_increase_hue_noeeprom();
+            } else {
+                rgblight_decrease_hue_noeeprom();
+            }
+        }
+    } else {                          // _QWERTY LAYER and others
+        if (clockwise) {
+            tap_code(KC_VOLU);
+        } else {
+            tap_code(KC_VOLD);
+        }
+    }
+  } else if (index == 1) {        // Encoder on slave side
+    if(IS_LAYER_ON(_LOWER)) {       // _LOWER LAYER
+      if (clockwise) {
+        tap_code(KC_PGDN);
+      } else {
+        tap_code(KC_PGUP);
+      }
+    } else if (IS_LAYER_ON(_MOD))     // _MOD LAYER
+    {
+        if ( get_mods() & MOD_MASK_SHIFT ) {
+            if (clockwise) {
+                rgblight_increase_speed_noeeprom();
+            } else {
+                rgblight_decrease_speed_noeeprom();
+            }
+        } else {
+            if (clockwise) {
+                rgblight_increase_val_noeeprom();
+            } else {
+                rgblight_decrease_val_noeeprom();
+            }
+        }
+    } else {                          // _QWERTY LAYER and others
+      if (clockwise) {
+          tap_code(KC_DOWN);
+      } else {
+          tap_code(KC_UP);    
+      }
+    }
+  }
+  return true;
```

Comment:
This will be overridden by the core behaviour otherwise.
```suggestion
  return false;
```